modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+
+
+
+[comment0]: <> (% ----------------------------------------------------------------------------------------------------------%)
+
+## PDE Constrained Inverse Optimization id=sec:invOptTheory
+
+Inverse optimization is a mathematical framework to infer model parameters by minimizing the misfit between the experimental and simulation observables.  In the optimization module, our model is a PDE describing the physics of the experiment.  We solve our physics PDE using the finite element method as implemented in MOOSE.  The physics of our problem constrains our optimization algorithm.  A PDE-constrained inverse optimization framework is formulated as an abstract optimization problem [!citet](biegler2003large):
+\begin{equation}\label{eq:optimization}
+   \min_{\mathbf{p}} J\left(\mathbf{u},\mathbf{p}\right);\quad~\textrm{subject to}~\mathbf{g}\left(\mathbf{u},\mathbf{p}\right)=\mathbf{0} ,
+\end{equation}
+where $J(\mathbf{u},\mathbf{p})$ is our objective function, which is a scalar measure of the misfit between experimental and simulated responses, along with any regularization [!citet](neumaier1998solving).  The constraint, $\mathbf{g}\left(\mathbf{u},\mathbf{p}\right)=\mathbf{0}$, consists of the PDEs governing the multiphysics phenomena simulated by MOOSE (e.g. coupled heat and elasticity equations), $\mathbf{p}$ contains model parameters (e.g. material properties or loads) and $\mathbf{u}$ contains simulated responses (e.g. temperature and displacement fields).  [eq:optimization] appears simple on the outset but is extremely difficult to solve. The solution space can span millions of degrees of freedom and the parameter space can also be very large. Finally, the PDEs can be highly nonlinear, time-dependent and tightly coupling complex phenomena across multiple physics.
+
+Optimization problems can be solved using either global (gradient-free) or local (gradient-based) approaches [!citet](aster2018parameter). Global approaches require a large number of iterations compared to gradient-based approaches (e.g. conjugate gradient or Newton-type methods), making the latter more suitable to problems with a large parameter space and computationally expensive models.  The [PETSc TAO](https://www.mcs.anl.gov/petsc/documentation/taosolvertable.html) optimization library [!citet](balay2019petsc) is used to solve [eq:optimization].  Optimization libraries like TAO require access to functions for computing the objective ($J$), gradient $\left(\mathrm{d}J/\mathrm{d}\mathbf{p}\right)$ and Hessian $\left(\mathrm{d}^2J/\mathrm{d}\mathbf{p}^2\right)$ or a function to take the action of the Hessian on a vector.  An objective function measuring the misfit or distance between the simulation and experimental data usually has the form
+\begin{equation}\label{eq:objective}
+J(\mathbf{u}, \mathbf{p}) = \frac{1}{2}\sum_i \left(\mathbf{u}_i - \mathbf{\bar u}_i\right)^2 + \frac{\rho}{2}\sum_i \mathbf{p}^2 ,
```

Comment:
Index needs to be explained: I guess `i` in the first term is node index or dof index; `i` in the second is the parameter index. `\mathbf{p}` in the second term possible should be `p_i`. Suggest to use different index notations for the two terms.

Reply:
done